cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -507,6 +507,10 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) caerrors.AutoscalerErr
 		a.AutoscalingContext.DebuggingSnapshotter.SetClusterNodes(l)
 	}
 
+	if err := a.processors.ProvisioningRequestProcessor.Process(); err != nil {
```

Comment:
ProvisioningRequestProcessor is aimed to process ProvisioningRequest conditions and check if they are expired, or the booking capacity mechanism is expired and update ProvisioningRequest conditions if needed.
This Processor should be called before ProvisioningRequestInjector (I haven't introduced it yet, it's will be in next PR), and ProvisioningRequestInjector will be a part of PodListProcessor (or it will be called just after PodListProcessor) and will be called after ProvisioningRequestFilter (which filter out real unscheduled pods that consume ProvisioningRequest from unscheduled pods to not trigger another scaleUp).

Currently we don't list ProvisioningRequests in the beginning of the loop (as we do for regular pods). I added a client to the processor and list ProvReqs inside the processor. That's why the input is empty.

Currently I'm adding support for check-capacity ProvisioningClass, but then I'll add support for atomicScaleUp ProvisioningClass. Each class will process ProvisioningRequest differently, so that's why ProvisioningRequestProcessor has a list of ProvisioningClassProcessor where each processor receives a list of ProvisioningRequests. 
Initially I thought to have one interface for ProvisioningRequestProcessor and ProvisioningClassProcessor and wanted to pass nil list to ProvisioningRequestProcessor in the RunOnce() function, but @kisieland asked to have an empty input instead, so I introduced two processors.